src/modules/hip/kernel/glitch.hpp

```diff
@@ -0,0 +1,299 @@
+    rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &dst_f8);
+
+    dstIdx += dstStridesNCH.y;
+    rpp_hip_load8_glitch(srcPtr + srcIdx, srcStridesNH.y, &glocSrc_f16, &srcRoi_i4, &dst_f8, 3, 1);
+    rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &dst_f8);
+
+    dstIdx += dstStridesNCH.y;
+    rpp_hip_load8_glitch(srcPtr + srcIdx, srcStridesNH.y, &blocSrc_f16, &srcRoi_i4, &dst_f8, 3, 2);
+    rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &dst_f8);
+}
+
+template <typename T>
+__global__ void glitch_pln3_pkd3_hip_tensor(T *srcPtr,
+                                            uint3 srcStridesNCH,
+                                            T *dstPtr,
+                                            uint2 dstStridesNH,
+                                            RpptChannelOffsets *rgbOffsetsPtr,
+                                            RpptROIPtr roiTensorPtrSrc)
+{
+
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStridesNCH.x);
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x * 3;
+
+    int4 srcRoi_i4 = *(reinterpret_cast<int4 *>(&roiTensorPtrSrc[id_z]));
+    RpptChannelOffsets rgbOffsets = rgbOffsetsPtr[id_z];
+
+    d_float24 dst_f24;
+    d_float16 rlocSrc_f16, glocSrc_f16, blocSrc_f16;
+
+    compute_glitch_locs_hip(id_x, id_y, rgbOffsets, roiTensorPtrSrc[id_z], &rlocSrc_f16, &glocSrc_f16, &blocSrc_f16);
+
+    rpp_hip_load8_glitch(srcPtr + srcIdx, srcStridesNCH.z, &rlocSrc_f16, &srcRoi_i4, &(dst_f24.f8[0]), 1, 0);
+
+    srcIdx += srcStridesNCH.y;
+    rpp_hip_load8_glitch(srcPtr + srcIdx, srcStridesNCH.z, &glocSrc_f16, &srcRoi_i4, &(dst_f24.f8[1]), 1, 0);
+
+    srcIdx += srcStridesNCH.y;
+    rpp_hip_load8_glitch(srcPtr + srcIdx, srcStridesNCH.z, &blocSrc_f16, &srcRoi_i4, &(dst_f24.f8[2]), 1, 0);
+
+    rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &dst_f24);
+}
+
+template <typename T>
+RppStatus hip_exec_glitch_tensor(T *srcPtr,
+                                 RpptDescPtr srcDescPtr,
+                                 T *dstPtr,
+                                 RpptDescPtr dstDescPtr,
+                                 RpptChannelOffsets *rgbOffsets,
+                                 RpptROIPtr roiTensorPtrSrc,
+                                 RpptRoiType roiType,
+                                 rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+    int globalThreads_x = dstDescPtr->strides.hStride;
+    int globalThreads_y = dstDescPtr->h;
+    int globalThreads_z = handle.GetBatchSize();
```

Comment:
Use the dstDescPtr->n for globalThreads_z instead of getting from handle since we are foreseeing some changes w.r.t handle in the future